src/xma/src/xmaapi/xmascaler.cpp

```diff
@@ -302,7 +303,7 @@ xma_scaler_session_create(XmaScalerProperties *sc_props)
     }
 
     //Obtain lock only for a) singleton changes & b) kernel_info changes
-    std::unique_lock<std::mutex> guard1(g_xma_singleton->m_mutex);
+    std::unique_lock<std::shared_timed_mutex> guard1(g_xma_singleton->m_mutex);
```

Comment:
```suggestion
    std::unique_lock guard1(g_xma_singleton->m_mutex);
```

Reply:
template type deduction not available in c++14